addons/mail/static/src/web/messaging_menu/messaging_menu.xml

```diff
@@ -64,38 +64,39 @@
                     body="failure.body"
-                    count="failure.notifications.length"
+                    counter="failure.notifications.length > 1 ? failure.notifications.length : undefined"
                     datetime="failure.datetime"
                     displayName="failure.modelName"
                     iconSrc="failure.iconSrc"
-                    isLast="displayedPreviews.length === 0 and failure_last"
+                    isLast="threads.length === 0 and failure_last"
                     hasMarkAsReadButton="true"
                     onClick="(isMarkAsRead) => isMarkAsRead ? this.cancelNotifications(failure) : this.onClickFailure(failure)"
                     onSwipeRight="hasTouch() ? { action: () => this.cancelNotifications(failure), icon: 'fa-times-circle', bgColor: 'bg-warning' } : undefined"
                 />
             </t>
-            <t t-foreach="displayedPreviews" t-as="preview" t-key="preview.id">
+            <t t-foreach="threads" t-as="thread" t-key="thread.localId">
+                <t t-set="message" t-value="thread.model === 'discuss.channel' ? thread.newestMessage : thread.newestNeedactionMessage"/>
                 <NotificationItem
-                    body="preview.message?.inlineBody"
-                    count="preview.count"
+                    body="message?.inlineBody"
+                    counter="thread.isChatChannel ? thread.message_unread_counter : thread.hasNeedactionMessages ? thread.message_needaction_counter : undefined"
                     datetime="message?.datetime"
-                    displayName="preview.thread.displayName"
-                    iconSrc="preview.imgUrl"
-                    hasMarkAsReadButton="preview.hasMarkAsReadButton"
-                    isLast="preview_last and failures.length === 0"
-                    onClick="(isMarkAsRead) => this.onClickPreview(isMarkAsRead, preview)"
-                    onSwipeRight="hasTouch() and preview.hasMarkAsReadButton ? { action: () => this.markAsRead(preview), icon: 'fa-check-circle', bgColor: 'bg-success' } : undefined"
-                    onSwipeLeft="hasTouch() and threadService.canUnpin(preview.thread) ? { action: () => this.threadService.unpin(preview), icon: 'fa-times-circle', bgColor: 'bg-danger' } : undefined"
+                    displayName="thread.displayName"
+                    iconSrc="thread.model === 'discuss.channel' ? thread.imgUrl : thread.newestNeedactionMessage.module_icon"
+                    hasMarkAsReadButton="thread.message_unread_counter > 0 or thread.hasNeedactionMessages"
+                    isLast="thread_last and failures.length === 0"
+                    onClick="(isMarkAsRead) => this.onClickThread(isMarkAsRead, thread)"
+                    onSwipeRight="hasTouch() and (thread.message_unread_counter > 0 or thread.hasNeedactionMessages) ? { action: () => this.markAsRead(thread), icon: 'fa-check-circle', bgColor: 'bg-success' } : undefined"
```

Comment:
This condition on `thread.message_unread_counter > 0 or thread.hasNeedactionMessages` appears so much and is so long to read that I would consider a getter.